### ISSUE TITLE
Red Cross user getting an error message when attempting to login using Facebook

### ISSUE THREAD

**Issue by OhMcGoo** [HTBox/allReady#311]
A Red Cross user is getting the following error message when attempting to login using Facebook:

Error
Given URL is not allowed by the Application configuration: One or more of the given URLs is not allowed by the App's settings. It must match the Website URL or Canvas URL, or the domain must be a subdomain of one of the App's domains.

**Comment by bcbeatty** [comment]
Separate Test Apps are need for urls, -d vs -mvp.
They _may_ also have to be declared as testers in the Roles section of the app properties page.

**Comment by OhMcGoo** [comment]
How do we get all potential users (the list is growing as I'm seeing our initial invite forwarded to others) assigned at testers?

**Comment by bcbeatty** [comment]
If we switch the app on facebook to the 'production' and be live. It won't require testers to be pre-defined. On the flip side it would allow anyone in with a social account, which is no different than the twitter and microsoft logins.

**Comment by bcbeatty** [comment]
But I'm not sure if it would allow for -d and -mvp sites to share the same app Ids, as well as when we do go in to production. We will need the test appIds.

**Comment by tonysurma** [comment]
@bcbeatty fine with flipping to production like you stated.  As App Ids are generated (really only need it for -mvp if it is a lot of work) I will update the deployment configuration once you get them (and share them in a non-public way with me :smile:)  Thanks

**Comment by tonysurma** [comment]
@bcbeatty do you still have perms to create these?

**Comment by bcbeatty** [comment]
Yes, I renamed "AllReady Test 1" to be "AllReady D", Created an "All Ready - MVP" and I changed the production to go to https://allready-mvp.azurewebsites.net/

**Comment by OhMcGoo** [comment]
We've had a second Red Cross user state that they can't login using Facebook. They were successful, however, logging in with Microsoft.

**Comment by OhMcGoo** [comment]
Users numbers three and four have expressed that they can't login using their Facebook credentials.

**Comment by bcbeatty** [comment]
I attempted to login and I get the following error: 
> Given URL is not allowed by the Application configuration: One or more of the given URLs is not allowed by the App's settings. It must match the Website URL or Canvas URL, or the domain must be a subdomain of one of the App's domains.

**Comment by bcbeatty** [comment]
Is MVP linking to Prod's Api key or the Test for MVP ?

**Comment by bcbeatty** [comment]
In the app Settings/Advanced Valid OAuth redirect URIs needed the base url (http and https)

**Comment by tonysurma** [comment]
@OhMcGoo so according to us devs this should be fixed, can you confirm?

**Comment by OhMcGoo** [comment]
This feature isn’t quite there. My password was rejected on my first attempt:



NOTE: This phrasing is confusing:  Passwords must have at least one non letter and non digit character.


My second attempt:

**Comment by tonysurma** [comment]
@OhMcGoo Looks like we missed some of what you typed.  However are you commenting on facebook login working or the direct username and password issue?  This is for facebook login and #330 is for 'regular' non social media logins

**Comment by OhMcGoo** [comment]
Let me try again.

The direct login feature isn’t quite there. My password was rejected on my first attempt (see Create account - 1st attempt).

NOTE: This phrasing is confusing:  Passwords must have at least one non letter and non digit character. A non later non digit is a symbol, right?

My second attempt brought about quite an astonishing display of characters! (see Create account - 2nd attempt).
![create account - 1st attempt](https://cloud.githubusercontent.com/assets/4147733/11139335/acce4ed2-8993-11e5-8830-7c92debfecdd.png)
![create account - 2nd attempt](https://cloud.githubusercontent.com/assets/4147733/11139336/ace4c644-8993-11e5-8aec-eae09d7f810c.png)

**Comment by OhMcGoo** [comment]
Still getting 'App Not Setup' error message when attempting to login via Facebook.

![app not setup](https://cloud.githubusercontent.com/assets/4147733/11139481/67f3b3f4-8995-11e5-8d98-2f67b9e2a3cd.png)

**Comment by bcbeatty** [comment]
This is because the app is still in test/development mode. We need your Facebook's profile Url to register you as a tester. Then Facebook will authenticate your login.

**Comment by OhMcGoo** [comment]
What about all our volunteers? Should we tell them NOT to use the Facebook option?

**Comment by tonysurma** [comment]
@bcbeatty @billwagner can we flip it to live please in Facebook?

**Comment by BillWagner** [comment]
@bcbeatty Ping me on slack when you have time. Let's get this working today.

**Comment by tonysurma** [comment]
just used a facebook account on -mvp and it all worked perfectly.  So unless @OhMcGoo sees anyone else with an error I think we can consider this closed and working.  Thanks @bcbeatty , @BillWagner , @dpaquette et al.

**Comment by OhMcGoo** [comment]
Success on my end. Thank you!
